watersheds/search/search.py

```diff
@@ -56,38 +56,50 @@ def get_mouth_source_segment(result, searcher):
 
   return mouth_segment, source_segment
 
-def bfs(result, searcher, mouth_segment, basin, t0):
+def bfs_river(basin, river, mouth_id, t0):
   # keep track of river geometries and basin ids
   wkts = []
   metadata = []
   basin_ids = set()
 
   # bfs from mouth segment
-  q = deque([mouth_segment])
+  q = deque([mouth_id])
   while q:
-    cur_segment = q.popleft()
+    cur_id = q.popleft()
+    if cur_id not in river.data_dict: continue
 
-    wkts.append(cur_segment['geometry'])
-    metadata.append([cur_segment['ORD_STRA'], cur_segment['ORD_CLAS'], cur_segment['ORD_FLOW']])
-    basin_ids.add(cur_segment['HYBAS_L12'])
+    wkts.append(river.get_geo_by_id(cur_id))
+    metadata.append(river.get_metadata_by_id(cur_id))
+    basin_ids.add(river.data_dict[cur_id]['HYBAS_L12'])
 
-    query = JLongPoint.newExactQuery('NEXT_DOWN', cur_segment['HYRIV_ID'])
-    hits = searcher.search(query, 4)
-    for hit in hits:
-      q.append(json.loads(hit.raw))
+    print(cur_id, river.next_river_id_dict[cur_id])
+    if cur_id not in river.next_river_id_dict: continue
+    for neighbour in river.next_river_id_dict[cur_id]:
+      q.append(neighbour)
 
   print("")
   print("Done BFS, Time:", time.time() - t0)
 
   return wkts, metadata, basin_ids
 
+def bfs_basin(basin, mouth_basin_id, t0):
```

Comment:
what is "t0" refering? Not used in this function.

Reply:
Removed